include/scheduler.h

```diff
@@ -51,13 +51,16 @@ namespace detail {
 		graph_generator& ggen;
 		graph_serializer& gsrlzr;
 
-		std::queue<scheduler_event> events;
+		std::queue<scheduler_event> available_events;
+		std::queue<scheduler_event> in_flight_events;
+		std::atomic_size_t in_flight_count = 0;
```

Comment:
This is only really needed by `scheduler::is_idle()`, which will be removed by #86 (only use case was the ad-hoc `slow_full_sync`). Depending on merge order, we can do without this variable entirely.